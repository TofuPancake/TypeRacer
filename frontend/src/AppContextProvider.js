import React, { useState, useEffect } from 'react';
import { auth } from './firebase';
import themeMusic from './sound/theme_music.mp3';

const AppContext = React.createContext();

function AppContextProvider({ children }) {
    const [firebaseUserIdToken, setFirebaseUserIdToken] = useState(null);
    const [username, setUsername] = useState('');
    const [lobbyid, setLobbyid] = useState('');
    const [lobby, setLobby] = useState(null);
    const audio = new Audio(themeMusic);
    audio.loop = true;
    audio.volume = 0.1;

    useEffect(() => {
        auth.onIdTokenChanged(async (auth) => {
            if (auth) {
                const token = await auth.getIdToken();
                setFirebaseUserIdToken(token);
            } else {
                setFirebaseUserIdToken(null);
            }
        });
    }, []);

    const context = {
        firebaseUserIdToken,
        username,
        setUsername,
        lobbyid,
        setLobbyid,
        lobby,
        setLobby,
        audio
    }

    return (
        <AppContext.Provider value={context}>
            {children}
        </AppContext.Provider>
    );
}

export {
    AppContext,
    AppContextProvider,
}
