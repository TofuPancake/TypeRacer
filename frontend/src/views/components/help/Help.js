import Modal from '../modal/Modal';
import { useState } from 'react';
import HelpIcon from '@material-ui/icons/Help';
import { Paper } from '@material-ui/core';

import styles from './Help.module.css';

export default function Help({ children }) {
    const [show, setShow] = useState(false);
    
    function handleCancelHelp() {
        setShow(false);
    }

    function handleShowHelp() {
        setShow(true);
    }


    return(
        show ? 
        <Modal style={{ width: '50%'}} dismissOnClickOutside={true} onCancel={handleCancelHelp}>
            <div className={styles.paper}>
                {children}
            </div>
        </Modal>
        :
        <div className={styles.button} onClick={handleShowHelp}><HelpIcon fontSize='large' color='action' /></div>
    );
}