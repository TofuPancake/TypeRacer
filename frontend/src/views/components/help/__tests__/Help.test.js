import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AppContextProvider } from '../../../../AppContextProvider';

import Help from '../Help';

test('help renders correctly', () => {
  const shallowRenderer = new ShallowRenderer();
  const snapshotComponent = shallowRenderer.render(
    <AppContextProvider>
      <Help />
    </AppContextProvider>
);
  expect(snapshotComponent).toMatchSnapshot();
});