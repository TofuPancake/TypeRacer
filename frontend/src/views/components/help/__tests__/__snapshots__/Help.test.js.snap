// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help renders correctly 1`] = `
<Context.Provider
  value={
    Object {
      "audio": <audio
        loop=""
        preload="auto"
        src="mock file"
      />,
      "firebaseUserIdToken": null,
      "lobby": null,
      "lobbyid": "",
      "setLobby": [Function],
      "setLobbyid": [Function],
      "setUsername": [Function],
      "username": "",
    }
  }
>
  <Help />
</Context.Provider>
`;
